From f724e53c404fc036809a2359a9c3332e562c4b40 Mon Sep 17 00:00:00 2001
From: Patrick Ohly <patrick.ohly@intel.com>
Date: Thu, 13 Jul 2017 17:44:35 +0200
Subject: [PATCH 1/2] meson: avoid unnecessary gdk dependency

gdk is only needed by the asb_plugin_font, which only gets built when
both appstream-build and fonts are enabled. Therefore checking for the
dependency should also be limited to that case, otherwise gdk has to
be provided during compilation also in cases where it is not used at
all.

Upstream-Status: Backport [https://github.com/hughsie/appstream-glib/pull/182]

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index b0912cc..c88bbdc 100644
--- a/meson.build
+++ b/meson.build
@@ -55,7 +55,6 @@ gmodule = dependency('gmodule-2.0', version : glib_ver)
 giounix = dependency('gio-unix-2.0', version : glib_ver)
 libarchive = dependency('libarchive')
 soup = dependency('libsoup-2.4', version : '>= 2.51.92')
-gdk = dependency('gdk-3.0')
 json_glib = dependency('json-glib-1.0')
 gdkpixbuf = dependency('gdk-pixbuf-2.0', version : '>= 2.31.5')
 libgcab = dependency('libgcab-1.0', required : false)
@@ -78,6 +77,7 @@ if get_option('enable-builder')
     conf.set('HAVE_ALPM', 1)
   endif
   if get_option('enable-fonts')
+    gdk = dependency('gdk-3.0')
     conf.set('HAVE_FONTS', 1)
     freetype = dependency('freetype2', version : '>= 9.10.0')
     fontconfig = dependency('fontconfig')
-- 
2.11.0

